locked-asset/energy-factory/src/migration.rs

```diff
@@ -55,10 +56,16 @@ pub trait SimpleLockMigrationModule:
         self.require_caller_old_factory();
         self.require_old_tokens_energy_was_updated(&original_caller);
 
+        let current_epoch = self.blockchain().get_block_epoch();
+
+        self.update_energy(&original_caller, |energy: &mut Energy<Self::Api>| {
+            for pair in initial_epoch_amount_pairs.pairs {
+                energy.update_after_unlock_any(&pair.amount, pair.epoch, current_epoch);
+            }
+        });
         self.update_energy(&original_caller, |energy: &mut Energy<Self::Api>| {
-            let current_epoch = self.blockchain().get_block_epoch();
-            for pair in epoch_amount_pairs.pairs {
-                energy.refund_after_token_unlock(&pair.amount, pair.epoch, current_epoch);
+            for pair in final_epoch_amount_pairs.pairs {
+                energy.add_after_token_lock(&pair.amount, pair.epoch, current_epoch);
             }
         });
```

Comment:
Do both `for`s into a single `update_energy`. This reads and writes the energy in storage twice

Reply:
Done.